libcloud/utils/misc.py

```diff
@@ -282,3 +292,28 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__repr__())
+
+
+def retry(timeout, retry_delay, backoff, func, *args, **kwargs):
+    """
+    Retry decorator helps to handle common exception.
+    """
+    timeout = DEFAULT_TIMEOUT if timeout is None else timeout
```

Comment:
Instead of doing that, you should simply use default function argument values:

```python
retry(timeout=DEFAULT_TIMEOUT, retry_delay=DEFAULT_SLEEP, backoff=DEFAULT_BACKOFF, func, *args, **kwargs):
```